### DISCUSSION TITLE
Customizing the OrderProcess

### DISCUSSION THREAD

**Discussion by agoransson** [vendurehq/vendure#2427] [Q&A] [answered]
Hi, I'm trying to figure out how to create a custom "check" before an order can be moved to checkout. In my store I want to have at maximum X number of orders waiting before a new order can be placed. It's for a small business where the stock is created on demand. So, we only want to have X number of orders available.

I was looking at the OrderProcess, to create a custom rule based on number of active orders (Waiting fulfilment) but I'm a but unsure how to start. Should my custom OrderProcess extend an already existing one, or should I create a whole new state in the order-state-machine?

Thanks

**Comment by michaelbromley** [comment, marked as answer]
Hi,

It sounds like you can keep the existing default OrderProcess but then add your own in addition, which defines the new "Checking" state.

In this new custom process you can override the state transition rules so that the order must proceed from `AddingItems -> Checking -> ArrangingPayment`, like in the example given here: https://beta-docs.vendure.io/guides/core-concepts/orders/#adding-a-new-state

**Reply by agoransson** [reply]
Oh, I see. I will try that.

Thank you Michael.

**Reply by agoransson** [reply]
Thanks Michael, it does work! Stripe however will still charge the customer it seems, even though the order didn't move to the expected state (as it did before my addition). Looking at the stripe plugin it will first attempt to charge the customer, and then transition the order.

Should I then modify the stripe plugin to first transition the order to "arranging payment" and if that succeeds I should proceed with charging the customer? Another alternative, I suppose, is if I transition the state from the webshop side, and then if that succeeds allow the customer to pay by stripe.

How would you approach this?
